File: internal/palette/palette.go
package palette

import "time"

type Palette struct {
	ID          int       `json:"id"`
	UserId      string    `json:"user_id"`
	Name        string    `json:"name"`
	Clover      string    `json:"clover"`
	Member      string    `json:"member"`
	Active      string    `json:"active"`
	Regular     string    `json:"regular"`
	Contributor string    `json:"contributor"`
	Addicted    string    `json:"addicted"`
	Insomniac   string    `json:"insomniac"`
	Nolifer     string    `json:"nolifer"`
	Birthday    string    `json:"birthday"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// PaletteStore defines database operations for palette's.
type PaletteStore interface {
	Get(paletteId int) (*Palette, error)
	GetByUser(userId string) ([]*Palette, error)
	Create(palette *Palette) error
	Update(palette *Palette) error
	Delete(paletteId int) error
}

File: old/models/Palette.js
module.exports = (sequelize, DataTypes) => sequelize.define('palette', {
  user_id: DataTypes.STRING,
  palette_name: DataTypes.STRING,
  clover: DataTypes.STRING,
  member: DataTypes.STRING,
  active: DataTypes.STRING,
  regular: DataTypes.STRING,
  contributor: DataTypes.STRING,
  addicted: DataTypes.STRING,
  insomniac: DataTypes.STRING,
  nolifer: DataTypes.STRING,
  birthday: DataTypes.STRING,
  created_at: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
});